spec/features/regen_api_key_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'spec_helper'
+
+describe 'regenerating api_key' do
+  
+  let(:user) { create(:user) }
+
+  before do
+    login_as user
+    visit settings_path
+  end
+
+  it 'should have a button for regenerating the API key' do
+    expect(page).to have_button 'Regenerate key'
+  end
+
+  it 'should warn the user about invalidating the API key' do
```

Comment:
I like that you're worried about testing this, but I'm ok with not testing the functionality since that's testing an alert.